File: backend/app/schemas/workspace.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.core.permissions import Role
from app.schemas.user import UserResponse


class WorkspaceBase(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: str = Field(..., min_length=2, max_length=100)
    description: Optional[str] = None
    industry: Optional[str] = None
    stage: Optional[str] = None
    website: Optional[str] = None
    region: Optional[str] = None
    currency: str = Field("USD", min_length=1, max_length=10)


class WorkspaceCreate(WorkspaceBase):
    pass


class WorkspaceUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: Optional[str] = Field(None, min_length=2, max_length=100)
    description: Optional[str] = None
    industry: Optional[str] = None
    stage: Optional[str] = None
    website: Optional[str] = None
    region: Optional[str] = None
    currency: Optional[str] = None


class WorkspaceResponse(WorkspaceBase):
    model_config = ConfigDict(from_attributes=True, extra="ignore")

    id: str
    owner_id: str
    created_at: datetime
    updated_at: datetime


class WorkspaceMemberResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    workspace_id: str
    user_id: str
    role: Role
    joined_at: datetime
    user: Optional[UserResponse] = None


class MemberInviteRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    email: EmailStr
    role: Role = Role.TEAM_MEMBER


class MemberUpdateRoleRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    role: Role

File: frontend/src/types/workspace.ts
import { Role, User } from './auth';

export interface Workspace {
  id: string;
  name: string;
  description?: string | null;
  industry?: string | null;
  stage?: string | null;
  website?: string | null;
  region?: string | null;
  currency: string;
  owner_id: string;
  created_at: string;
  updated_at: string;
}

export interface WorkspaceMember {
  id: string;
  workspace_id: string;
  user_id: string;
  role: Role;
  joined_at: string;
  user?: User;
}

export interface WorkspaceCreatePayload {
  name: string;
  description?: string;
  industry?: string;
  stage?: string;
  website?: string;
  region?: string;
  currency?: string;
}

export interface MemberInvitePayload {
  email: string;
  role: Role;
}
